plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.test.ts

```diff
@@ -100,6 +104,7 @@ function setupStubs(projects: Project[], baseUrl: string) {
                   },
                 ],
               },
+              defaultBranch: defaultBranch,
```

Comment:
@davelil4 what version of Bitbucket server are you testing this on. I am using version 7.21 and do not have this field available.

I think you'll want to mark this as an optional field and fall back to retrieving it via the default branch API which has been around for a while, otherwise this might cause some issues for people who haven't upgraded bitbucket.

Reply:
@bforbis I do not have access to a bitbucket server instance anymore (I did at the job I was working at, but was thinking about trying to spin up a trial instance if I can) so I was just going on their current documentation, and building test cases from that. I will fix this as soon as I get the chance to call the API to grab the default branch of the repository if the field is missing.

Reply:
Ok, that would be this API

**rest/api/latest/projects/:project/repos/:slug/default-branch**

Which would return a response similar to
```json
{
    "id": "refs/heads/master",
    "displayId": "master",
    "type": "BRANCH",
    "latestCommit": "0f2f3ae484054696568bf4560ba4da280f7df82a",
    "latestChangeset": "0f2f3ae484054696568bf4560ba4da280f7df82a",
    "isDefault": true
}
```

Reply:
@bforbis Sounds good, thank you for providing the API and its sample payload. I will implement over this weekend or early next week ASAP.

Reply:
@bforbis I've just found that this API endpoint is deprecated in V8 of BBS. I can't find online as to whether the endpoint works or not for version 8. Do you know if this endpoint will work for both versions?

Reply:
Oh that's interesting. I do not have access to Bitbucket Server version 8, so there isn't an easy way for me to check.

Though I imagine that by version 8, you will always have the `defaultBranch` attribute in the getRepo API, so if you code it to only hit that API when that field is not available, you should likely be good.